### PR TITLE
refactor: upgrade go version to 1.18.6

### PR DESCRIPTION
**Pull request by aahmed7**
This upgrades go version to 1.18

related to: https://github.com/jenkins-x/jx/issues/8283

- [X] Change the version in go.mod file.
- [X] Run go mod tidy and ensure that it does not error out.
- [X] Change GO_VERSION in the makefile to the version you are upgrading.
- [X] Check if there are any references to an older version and replace those.
- [X] Verify that
  - [X] The build works by running `make build`
  - [X] Tests are working, by running `make test`

### CODE REVIEW

**Comment by aahmed7** [comment]
ran a basic runtime test as well apart from `make test` 
```
$ ./build/jx-verify version
version: -dev+932a56f
```

File: Makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@ ORG := jenkins-x
 ORG_REPO := $(ORG)/$(NAME)
 RELEASE_ORG_REPO := $(ORG_REPO)
 ROOT_PACKAGE := github.com/$(ORG_REPO)
-GO_VERSION := 1.17.9
+GO_VERSION := 1.18.6
```
Comment (ankitm123) [inline Makefile]: Replace GO_VERSION with this section: https://github.com/jenkins-x-plugins/jx-pipeline/blob/89593c6e1e9d2c01b8a26296f53b6fdedcc48410/Makefile#L16-L22.

**Comment by ankitm123** [comment]
Remember to upgrade the k8s client libraries to 0.22.X here: https://github.com/jenkins-x-plugins/jx-verify/blob/932a56fdb7e7b77aecd5647c2639446fecbca829/go.mod#L87-L91

**Comment by aahmed7** [comment]
@ankitm123  sure

**Review by ankitm123 — APPROVED** [review]

**Comment by ankitm123** [comment]
@aahmed7 thanks for doing this. We will be participating in hacktoberfest, and we will most likely have lot of issues for people to contribute to :)

**Pull request merged**